packages/flutter/lib/src/widgets/animated_scroll_view.dart

```diff
@@ -148,12 +234,34 @@ class AnimatedList extends _AnimatedScrollView {
     return context.findAncestorStateOfType<AnimatedListState>();
   }
 
+  // Helper method to compute the actual child count when taking separators into account.
+  static int _computeChildCountWithSeparators(int itemCount) {
+    if (itemCount == 0) {
+      return 0;
+    }
+    return itemCount * 2 - 1;
+  }
+
   @override
   AnimatedListState createState() => AnimatedListState();
 }
 
-/// The [AnimatedListState] for [AnimatedList], a scrolling list container that animates items when they are
-/// inserted or removed.
+/// Signature for the builder callback used by [AnimatedList.separated] to
+/// build its animated separators.
+///
+/// The [context] argument is the build context where the widget will be
+/// created, the [index] is the index of the item of the separator to be built,
+/// and the [animation] is an [Animation] that should be used to animate an entry
+/// transition for the widget that is built.
+///
+/// See also:
+///
+/// * [AnimatedRemovedItemBuilder], a builder that is used for removing items with
+///   animations instead of adding them.
+typedef AnimatedSeparatorBuilder = Widget Function(BuildContext context, int index, Animation<double> animation);
```

Comment:
It looks like this is the same signature as AnimatedItemBuilder, do you think the separator builder could use the same?

Reply:
My thought was to separate them semantically with different naming for item and separator. It might be nicer to only define unique signatures? I updated the docstring for `AnimatedItemBuilder` accordingly. I also thought about a different name for it, but don't want to make this a breaking change :)